panel/reactive.py

```diff
@@ -399,22 +399,26 @@ class Reactive(Syncable, Viewable):
     # Public API
     #----------------------------------------------------------------
 
-    def link(self, target, callbacks=None, bidirectional=False,  **links):
+    def link(self, target: param.Parameterized,
+        callbacks: Optional[Dict[str, Union[str, Callable]]]=None,
+        bidirectional: bool=False,  **links):
         """
-        Links the parameters on this object to attributes on another
-        object in Python. Supports two modes, either specify a mapping
-        between the source and target object parameters as keywords or
+        Links the parameters on this `Reactive` object to attributes on the
+        target `Parameterized` object.
+        
+        Supports two modes, either specify a
+        mapping between the source and target object parameters as keywords or
         provide a dictionary of callbacks which maps from the source
         parameter to a callback which is triggered when the parameter
         changes.
 
         Arguments
         ---------
-        target: object
+        target: param.Parameterized
           The target object of the link.
-        callbacks: dict
+        callbacks: dict | None
```

Comment:
From python 3.10 you can use `dict | None` instead of `Optional[None]`. I believe its more future proof to us the modern version in the docstrings. But keep the latter in the code for compatibility with python 3.6 for now.

Reply:
Happy to drop 3.6 asap but I guess that's not enough.

Reply:
I don't think there is a problem now. But the did python 3.10 is the minimum supported version the the type hints could be updated.

Reply:
I would prefer to see the pipe notation from the start, and this will also avoid needing to update the type hints in a couple of years. 

Using the pipe notation is possible from python 3.7 by adding `from __future__ import annotations`. I see no harm in doing this as bokeh is pinned to at least 2.4, and that version does not support python 3.6 [[ref](https://github.com/bokeh/bokeh/wiki/BEP-9:-Downstream-Version-Support)]. Furthermore, if you look at the bokeh's code it also imports from the future.